cmd/push/main.go

```diff
@@ -95,10 +97,34 @@ func (t tool) push(verbose bool, src string, pkgs []string, host, rel string) er
 
 }
 
+// See print_rsync_version https://git.samba.org/?p=rsync.git;a=blob;f=options.c
+var rsyncVersionRegexp = regexp.MustCompile(`^rsync\s+version\s+(\d+)\.(\d+)`)
+
+// checkRsyncVersion checks whether the rsync version is supported.
+func checkRsyncVersion(version []byte) bool {
+	matches := rsyncVersionRegexp.FindSubmatch(version)
+	if matches == nil {
+		return false
+	}
+	major, err := strconv.Atoi(string(matches[1]))
+	if err != nil {
+		return false
+	}
+	minor, err := strconv.Atoi(string(matches[2]))
+	if err != nil {
+		return false
+	}
+	// --info=progress2 has been introduced in 3.1.0
```

Comment:
I'd prefer instead:

- `getRsyncVersion() (int, int)` that returns major and minor version, and 0, 0 if none.
- in detect, return `rsyncOld` or `rsyncRecent` based on the minimum version
- On rsyncOld, skip the flag --info-progress2.